File: backend/src/main/java/com/school/saas/module/school/SchoolService.java
package com.school.saas.module.school;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.UUID;

public interface SchoolService {

    SchoolDTO create(CreateSchoolRequest request);

    SchoolDTO update(UUID id, UpdateSchoolRequest request);

    SchoolDTO getById(UUID id);

    SchoolDetailDTO getDetailById(UUID id);

    Page<SchoolDTO> getAll(Pageable pageable);

    List<SchoolDTO> getAllActive();

    void delete(UUID id);

    void activate(UUID id);

    void deactivate(UUID id);
}

File: frontend/src/api/school.service.ts
import apiClient from './client'
import type { ApiResponse, PageResponse, School } from '@/types'

export interface CreateSchoolRequest {
  name: string
  address: string
  email: string
  phone: string
  logoUrl?: string
}

export interface UpdateSchoolRequest {
  name?: string
  address?: string
  email?: string
  phone?: string
  logoUrl?: string
  active?: boolean
}

export const schoolService = {
  getAll: async (page = 0, size = 10) => {
    const response = await apiClient.get<PageResponse<School>>('/schools', {
      params: { page, size },
    })
    return response.data
  },

  getById: async (id: string) => {
    const response = await apiClient.get<ApiResponse<School>>(`/schools/${id}`)
    return response.data
  },

  create: async (data: CreateSchoolRequest) => {
    const response = await apiClient.post<ApiResponse<School>>('/schools', data)
    return response.data
  },

  update: async (id: string, data: UpdateSchoolRequest) => {
    const response = await apiClient.put<ApiResponse<School>>(`/schools/${id}`, data)
    return response.data
  },

  delete: async (id: string) => {
    await apiClient.delete(`/schools/${id}`)
  },
}
